reports/GitalyFail/GitalyFail.tla

```diff
@@ -0,0 +1,137 @@
+------------------------------ MODULE GitalyFail ------------------------------
+(* 
+    MIT License Copyright (c) 2020 CQFN
+    https://github.com/cqfn/degitx/blob/master/LICENSE
+    
+    DeGit Tech Report. 
+    Appendix 1. Gitaly fails explore.
+           
+ *)
+===============================================================================
+CONSTANT RM
+
+VARIABLES rmState, tmState, tmPrepared, 
+    msgs, rmGenerations, tmGeneration, 
+
+Messages == [type: {"Prepared"}, rm: RM] \union [type:{"Commit", "Abort"}]
+
+
+TPTypeOK == /\ rmState \in [RM -> {"working", "prepared", "pre-committed", "committed", "aborted"}]
+            /\ tmState \in {"init", "done"}
+            /\ tmPrepared \subseteq RM
+            /\ msgs \subseteq Messages
+            
+TPInit ==   /\ rmState = [r \in RM |-> "working"]
+            /\ tmState = "init"
+            /\ tmPrepared = {}
+            /\ msgs = {}            
+            /\ rmGenerations = [r \in RM |-> 0]
+            /\ tmGeneration = 0
```

Comment:
What is `tmGeneration` here? In Gitaly we don't have generation field for TM (@ivanbukhtiyarov is it correct?)

Reply:
@g4s8 @IlyaMoskva yes, Gitaly don't have generation field for TM or anything else. Only Praefect works with "generations" to monitor gitaly nodes state. Nodes that successfully committed should have their generations incremented. Nodes that did not successfully commit or did not participate should remain on their existing generation.

Reply:
@g4s8, Here tmGeneration means the version of the repository's branch. To compare it with nodes later. Maybe, it is not suitable name.

Reply:
@ivanbukhtiyarov can you please check where exactly the global generation is updated for repository shard? @IlyaMoskva I think we can use something like `expectedGeneration` or similar for thus variable.

Reply:
@g4s8 @IlyaMoskva if you mean virtual storage generation, praefect works with any generation incrementation (repositories or storage_repositories tables) in [`IncrementGeneration`](https://gitlab.com/gitlab-org/gitaly/-/blob/13-9-stable/internal/praefect/datastore/repository_store.go#L165) method that called in transaction finalizer.